modules/planning/common/obstacle_blocking_analyzer.cc

```diff
@@ -35,14 +35,18 @@ using apollo::hdmap::HDMapUtil;
 // TODO(all): if possible, transform as many function parameters into GFLAGS.
 bool IsBlockingObstacleToSidePass(
     const Frame& frame, const Obstacle* obstacle,
-    double block_obstacle_min_speed, double min_front_sidepass_distance,
-    bool enable_obstacle_blocked_check) {
+    double block_obstacle_min_speed,
+    double min_front_sidepass_distance,
+    bool enable_obstacle_blocked_check,
+    double* const distance_between_adc_and_obstacle) {
```

Comment:
It is better to re-design the function to make it do one thing at a time. Getting the distance and checking blocking seem to be two tasks. Can they be separated into two functions?

Reply:
Getting the distance seems more like a "side" task because it is only one-line of code. We can, of course, separate them into two. But I'm a little worried of the consistency problem: there can be multiple different definitions of the distance between adc and obstacle: the spacing between centers, the spacing between edges, etc.